VAMobile/android/app/src/main/java/gov/va/mobileapp/native_modules/CustomTabsIntentModule.kt

```diff
@@ -84,13 +86,27 @@ class CustomTabsIntentModule(private val context: ReactApplicationContext) :
                             }
                             .build()
 
-            // Check default browser to prevent Firefox login issue (Android only)
-            val browserIntent = Intent("android.intent.action.VIEW", Uri.parse("https://"));
-            val resolveInfo = context.packageManager.resolveActivity(browserIntent, PackageManager.MATCH_DEFAULT_ONLY);
-            val packageName = resolveInfo?.activityInfo?.packageName;
-            if (packageName != null && packageName.contains("firefox")) {
-                // Default browser is Firefox. Need flag for login to succeed
-                customTabsIntent.intent.addFlags(FLAG_ACTIVITY_NEW_TASK)
+            // Get full list of installed packages that handle https URLs
+            // https://developer.chrome.com/docs/android/custom-tabs/howto-custom-tab-check/
+            val intent = Intent(ACTION_VIEW, Uri.parse("https://www.example.com"))
+            val intentHandlers = context.packageManager.queryIntentActivities(intent, PackageManager.MATCH_ALL)
+            val intentHandlerPackages = intentHandlers.map {
+                it.activityInfo.packageName
+            }.toTypedArray().asList()
+
+            // From the above list, find a package that supports Custom Tabs. If no installed
+            // packages support Custom Tabs, packageName will be set to null here
+            val packageName = CustomTabsClient.getPackageName(context, intentHandlerPackages)
```

Comment:
Not clear by the comment: this will preferentially choose their default browser as described by the PR write-up, right? Wanted to double check it wouldn't arbitrarily just pull the first in the list and end up doing it alphabetically or whatever the prior list comes as.

Reply:
Answered my own question following the Chrome Dev link above--yes, getPackageName does preferentially choose default unless told not to.

Defer to you if it's worth clarifying the comment.

Reply:
Exactly, the third argument to getPackageName determines that. I tested a bit with multiple browsers installed. My choice of default browser was respected as long as it supported custom tabs. So it seems to work as advertised. QA will test further to make sure